include/openssl/provider.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef OSSL_PROVIDER_H
+# define OSSL_PROVIDER_H
+
+# include <openssl/core.h>
+
+# ifdef __cplusplus
+extern "C" {
+# endif
+
+/* Load and unload a provider */
+OSSL_PROVIDER *OSSL_load_provider(OPENSSL_CTX *, const char *name);
+int OSSL_unload_provider(OSSL_PROVIDER *prov);
+
+const OSSL_ITEM *OSSL_get_provider_param_types(OSSL_PROVIDER *prov);
```

Comment:
OSSL_PROVIDER_get_param_types

Reply:
This function doesn't manipulate the `OSSL_PROVIDER` type, it only uses it.

Reply:
Again in this case too I agree with @levitte.

Reply:
OTOH, looking at the implementation of this function it does actually read "private" data of the prov object in order to do its thing, so maybe for this one (and therefore the next one) @t-j-h is right.

Reply:
Hmm...  okie